diff --git a/brittany.cabal b/brittany.cabal
index a33136b..bde77f6 100644
--- a/brittany.cabal
+++ b/brittany.cabal
@@ -93,7 +93,6 @@ library {
   build-depends:
     { base >=4.12 && <4.15
     , ghc >=8.6.1 && <8.11
-    , ghc-paths >=0.1.0.9 && <0.2
     , ghc-exactprint >=0.5.8 && <0.6.4
     , transformers >=0.5.2.0 && <0.6
     , containers >=0.5.7.1 && <0.7
@@ -173,7 +172,6 @@ test-suite unittests
     { brittany
     , base
     , ghc
-    , ghc-paths
     , ghc-exactprint
     , transformers
     , containers
@@ -244,7 +242,6 @@ test-suite littests
     { brittany
     , base
     , ghc
-    , ghc-paths
     , ghc-exactprint
     , transformers
     , containers
