diff --git a/ghc-check.cabal b/ghc-check.cabal
index 2025482..6e80e14 100644
--- a/ghc-check.cabal
+++ b/ghc-check.cabal
@@ -23,7 +23,6 @@ library
                        directory,
                        filepath,
                        ghc,
-                       ghc-paths,
                        process,
                        safe-exceptions,
                        template-haskell,
diff --git a/src/GHC/Check/Util.hs b/src/GHC/Check/Util.hs
index 8c6e95d..72df3ac 100644
--- a/src/GHC/Check/Util.hs
+++ b/src/GHC/Check/Util.hs
@@ -11,7 +11,6 @@ import           Data.Maybe
 import           Data.Version
 import           GHC (Ghc, gcatch)
 import           GHC.Exts                   (IsList (fromList), toList)
-import qualified GHC.Paths
 import           Language.Haskell.TH
 import           Language.Haskell.TH.Syntax as TH
 import           System.Environment (lookupEnv)
