diff --git a/ghc-exactprint.cabal b/ghc-exactprint.cabal
index 12ba9a9..4cafde6 100644
--- a/ghc-exactprint.cabal
+++ b/ghc-exactprint.cabal
@@ -112,7 +112,6 @@ library
                      , directory >= 1.2
                      , filepath  >= 1.4
                      , ghc       >= 7.10.2
-                     , ghc-paths >= 0.1
                      , mtl       >= 2.2.1
                      , syb       >= 0.5
                      , free      >= 4.12
@@ -198,7 +197,6 @@ Test-Suite test
                      , directory >= 1.2
                      , filepath  >= 1.4
                      , ghc  >= 7.10.2
-                     , ghc-paths  >= 0.1
                      , mtl        >= 2.2.1
                      , syb        >= 0.5
                      , silently   >= 1.2
@@ -232,7 +230,6 @@ executable roundtrip
                , filepath
                , ghc
                , ghc-exactprint
-               , ghc-paths
                , syb
                , temporary
                , time
@@ -278,7 +275,6 @@ executable prepare-hackage
                  , filemanip
                  , filepath
                  , ghc       >= 7.10.2
-                 , ghc-paths >= 0.1
                  , HUnit
                  , text   >= 1.2.2
                  , turtle >= 1.3.0
diff --git a/src/Language/Haskell/GHC/ExactPrint/Parsers.hs b/src/Language/Haskell/GHC/ExactPrint/Parsers.hs
index 6809050..62b716a 100644
--- a/src/Language/Haskell/GHC/ExactPrint/Parsers.hs
+++ b/src/Language/Haskell/GHC/ExactPrint/Parsers.hs
@@ -55,7 +55,7 @@ import Control.Monad.RWS
 import Data.Data (Data)
 #endif
 
-import GHC.Paths (libdir)
+import System.Environment (lookupEnv)
 
 import qualified ApiAnnotation as GHC
 import qualified DynFlags      as GHC
@@ -284,9 +284,10 @@ parseModuleApiAnnsWithCpp cppOptions file = ghcWrapper $ do
 
 -- | Internal function. Default runner of GHC.Ghc action in IO.
 ghcWrapper :: GHC.Ghc a -> IO a
-ghcWrapper =
+ghcWrapper ghc = do
+  libdir <- lookupEnv "NIX_GHC_LIBDIR"
   GHC.defaultErrorHandler GHC.defaultFatalMessager GHC.defaultFlushOut
-    . GHC.runGhc (Just libdir)
+    $ GHC.runGhc libdir ghc
 
 -- | Internal function. Exposed if you want to muck with DynFlags
 -- before parsing.
